inc/Engine/CriticalPath/CriticalCSSSubscriber.php

```diff
@@ -707,19 +625,34 @@ public function async_css( $buffer ) {
 				continue;
 			}
 
-			$preload = str_replace( 'stylesheet', 'preload', $tags_match[1][ $i ] );
-			$onload  = preg_replace( '~' . preg_quote( $tags_match[3][ $i ], '~' ) . '~iU', ' as="style" onload=""' . $tags_match[3][ $i ] . '>', $tags_match[3][ $i ] );
-			$tag     = str_replace( $tags_match[3][ $i ] . '>', $onload, $tag );
-			$tag     = str_replace( $tags_match[1][ $i ], $preload, $tag );
-			$tag     = str_replace( 'onload=""', 'onload="this.onload=null;this.rel=\'stylesheet\'"', $tag );
-			$buffer  = str_replace( $tags_match[0][ $i ], $tag, $buffer );
-
+			$tag        = str_replace( $tags_match[3][ $i ], 'as="style" onload="" ' . $tags_match[3][ $i ], $tag );
+			$tag        = str_replace( $tags_match[4][ $i ], 'media="print"', $tag );
+			$tag        = str_replace( 'onload=""', 'onload="this.media=\'all\'"', $tag );
```

Comment:
What if the original `media` attribute value is not `all`? We should preserve the original value.

It's also possible the `media` attribute is not set. In that case, we should add it, and use `all` by default on the `onload`